eif_utils/src/lib.rs

```diff
@@ -611,6 +613,126 @@ impl SignCertificateInfo {
     }
 }
 
+/// PCR Signature verifier that checks the validity of
+/// the certificate used to sign the enclave
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct PcrSignatureChecker {
+    signing_certificate: Vec<u8>,
+    signature: Vec<u8>,
+}
+
+impl PcrSignatureChecker {
+    pub fn new(pcr_signature: &PcrSignature) -> Self {
+        PcrSignatureChecker {
+            signing_certificate: pcr_signature.signing_certificate.clone(),
+            signature: pcr_signature.signature.clone(),
+        }
+    }
+
+    /// Reads EIF section headers and looks for a signature.
+    /// Seek to the signature section, if present, and save the certificate and signature
+    pub fn from_eif(eif_path: &str) -> Result<Self, String> {
+        let mut signing_certificate = Vec::new();
+        let mut signature = Vec::new();
+
+        let mut curr_seek = 0;
+        let mut eif_file =
+            File::open(eif_path).map_err(|e| format!("Failed to open the EIF file: {:?}", e))?;
+
+        // Skip header
+        let mut header_buf = vec![0u8; EifHeader::size()];
+        eif_file
+            .read_exact(&mut header_buf)
+            .map_err(|e| format!("Error while reading EIF header: {:?}", e))?;
+
+        curr_seek += EifHeader::size();
+        eif_file
+            .seek(SeekFrom::Start(curr_seek as u64))
+            .map_err(|e| format!("Failed to seek file from start: {:?}", e))?;
+
+        let mut section_buf = vec![0u8; EifSectionHeader::size()];
+
+        // Read all section headers and skip if different from signature section
+        while eif_file
+            .read_exact(&mut section_buf)
+            .map_err(|e| format!("Error while reading EIF section header: {:?}", e))
+            .is_ok()
+        {
+            let section = EifSectionHeader::from_be_bytes(&section_buf)
+                .map_err(|e| format!("Error extracting EIF section header: {:?}", e))?;
+            curr_seek += EifSectionHeader::size();
+
+            if section.section_type == EifSectionType::EifSectionSignature {
+                let mut buf = vec![0u8; section.section_size as usize];
+                eif_file
+                    .seek(SeekFrom::Start(curr_seek as u64))
+                    .map_err(|e| format!("Failed to seek after EIF section header: {:?}", e))?;
+                eif_file.read_exact(&mut buf).map_err(|e| {
+                    format!("Error while reading signature section from EIF: {:?}", e)
+                })?;
+
+                // Deserialize PCR signature structure and save certificate and signature
+                let des_sign: Vec<PcrSignature> = from_slice(&buf[..])
```

Comment:
Aren't these 2 equivalent?

```suggestion
                let des_sign: Vec<PcrSignature> = from_slice(&buf)
```